model/src/DA/Finance/Types.daml

```diff
@@ -63,3 +63,19 @@ data MasterAgreement = MasterAgreement
     party2 : Party
       -- ^ Allows to specify choices of the second counterparty.
   deriving (Eq, Show)
+
+-- | A record for templates which can be locked. Also includes a set of
+--  parties which may unlock a locked template.
+data Lock = Lock
+  with
+    by : Party
+      -- ^ The party which performed the locking action.
+    at : Time
```

Comment:
we kind of have the time the lock was established via the timestamp of the transaction. while it's not readily available through the JSON API (only through the gRPC one), I also don't see an immediate use case where this would be important or required. In the spirit of minimalism, maybe leave it out for now

Reply:
My thoughts around the time the lock was established are purely from a business perspective - as a user (or a business), knowing if a deposit has been locked for a day or a year would be useful information to have. Also, I was considering there may be use cases where we'd like to automatically release a lock after a certain time period and thus knowing when the deposit was locked would facilitate that.

As you say, it's not an immediate use case so I can drop it.